test/system/src/main/java/io/pravega/test/system/framework/services/docker/DockerBasedService.java

```diff
@@ -94,31 +115,28 @@ public boolean isRunning() {
                     }
                 }
             }
-            if (!serviceList.isEmpty()) {
-                long replicas = Exceptions.handleInterruptedCall(
-                        () -> dockerClient.inspectService(serviceList.get(0).id()).spec().mode().replicated().replicas());
-                log.info("Replicas {}", replicas);
-                log.info("Task running count {}", taskRunningCount);
-                if (((long) taskRunningCount) == replicas) {
-                    return true;
-                }
+            long replicas = getReplicas();
+            log.info("Replicas {}", replicas);
+            log.info("Task running count {}", taskRunningCount);
+            if (((long) taskRunningCount) == replicas) {
+                return true;
             }
         } catch (DockerException e) {
             log.error("Unable to list docker services", e);
         }
-        return value;
+        log.info("Service is not synced");
+        return false;
     }
 
     CompletableFuture<Void> waitUntilServiceRunning() {
-        log.debug("IS RUNNING {}", isRunning());
-        return Futures.loop(() -> !isRunning(), //condition
+        log.debug("IS RUNNING {}", isSynced());
```

Comment:
this log statement can be improved to include the service name too.

Reply:
Fixed